cmd/agent/app/flare.go

```diff
@@ -86,35 +87,73 @@ var flareCmd = &cobra.Command{
 	},
 }
 
-func readProfileData(pdata *flare.ProfileData) error {
+type profileCollector func(prefix, debugURL string, cpusec int, target *flare.ProfileData) error
+
+func readProfileData(pdata *flare.ProfileData, seconds int, collector profileCollector) error {
 	prevSettings, err := setRuntimeProfilingSettings()
 	if err != nil {
 		return err
 	}
 	defer resetRuntimeProfilingSettings(prevSettings)
 
+	agentCollectors := []struct {
```

Comment:
I like this 🍰

Reply:
Thank you! I decided against collecting these profiles concurrently, since this may have effect on the accuracy actual performance data.